### ISSUE TITLE
Extra Character Row for full Screen TUIs ("Scroll 0/1")

### ISSUE THREAD

**Issue by KyleKing** [zellij-org/zellij#952]
**Description**

When launching a full screen TUI in Zellij, the screen itsefl scrolls, when it should have a 100% fill. This doesn't appear to be terminal emulator or library-specific because I tried TUI applications implemented in several languages ([bpytop](https://github.com/aristocratos/bpytop), [lazygit](https://github.com/jesseduffield/lazygit), htop, etc.) in iTerm2 and Alacritty

**Further information**

To reproduce, launch zellij, launch any fullscreen TUI (i.e. `lazygit` or `htop`), then attempt to scroll and you should see the whole interface move up or down one row when it instead should fill the screen and not move

Demo with asciinema:

[![asciicast](https://asciinema.org/a/457069.svg)](https://asciinema.org/a/457069)

Logs from testing with lazygit:

[zellij-11.log](https://github.com/zellij-org/zellij/files/7739241/zellij-11.log)
[zellij.log](https://github.com/zellij-org/zellij/files/7739242/zellij.log)

**Device information**

`zellij --version`: zellij 0.22.1
`tput lines`: 58
`tput cols`: 192
`uname -av` or `ver`(Windows): Darwin MacBook-Pro.local 19.6.0 Darwin Kernel Version 19.6.0: Sun Nov 14 19:58:51 PST 2021; root:xnu-6153.141.50~1/RELEASE_X86_64 x86_64
`alacritty --version`: `alacritty 0.9.0 (fed349a)`
`lazygit --version`: commit=, build date=, build source=homebrew, version=0.31.4, os=darwin, arch=amd64
(Reproducible in latest versions of iTerm2 or Kitty and with htop, bpytop, etc.)

**Experimentation**

I tried: `zellij options --no-pane-frames` and/or the most generic layout possible and the error persisted:

```yaml
---
template:
  direction: Horizontal
  parts:
    - direction: Vertical
      no_pane_frames: true
```

**Comment by KyleKing** [comment]
This is a blocker for me, so tried to take a look in more depth

This appears to be introduced in Zellij 0.18 and isn't present in 0.17 or 0.16. I think it might be related to this change (#746), but I don't know Go or the codebase. This was the full diff between releases: https://github.com/zellij-org/zellij/compare/1868816791caca222f8a17ab96e3d514dc2ae79f...50452b53cbd9f50001a9662fba3faef393b73c8b

**Comment by KyleKing** [comment]
Okay, so reverting https://github.com/zellij-org/zellij/commit/fb8e44b8a8454887e61b8a331fbd7ea4a0a817c5 by restoring `self.viewport.clear()` fixes this issue. So that is the relevant code, but I really don't know what the right behavior should be and what the changes in that commit were fixing

But this is progress at least!

When this function:

https://github.com/zellij-org/zellij/blob/0863e74d10529358c82d0f3465c184ef38b6727d/zellij-server/src/panes/grid.rs#L909-L926

Is replaced with:

```rust
pub fn fill_viewport(&mut self, character: TerminalCharacter) {
    self.viewport.clear();
    for _ in 0..self.height {
        let columns = VecDeque::from(vec![character; self.width]);
        self.viewport.push(Row::from_columns(columns).canonical());
    }
    self.output_buffer.update_all_lines();
}
```

**Comment by tlinford** [comment]
@KyleKing you were spot on, hope to get this sorted pretty soon

**Comment by tlinford** [comment]
fixed in #1032
